lib/rules/selector-max-attribute/index.mjs

```diff
@@ -92,8 +90,21 @@ const rule = (primary, secondaryOptions) => {
 			if (!isStandardSyntaxRule(ruleNode)) return;
 
 			flattenNestedSelectorsForRule(ruleNode, result).forEach(({ selector, resolvedSelectors }) => {
-				resolvedSelectors.forEach((resolvedSelector) => {
-					checkSelector(resolvedSelector, selector, ruleNode);
+				resolvedSelectors.walk((childSelector) => {
+					if (childSelector.type !== 'selector') return;
+
+					if (isRoot(childSelector.parent)) {
+						checkSelector(childSelector, selector, ruleNode);
+
+						return;
+					}
+
+					if (
+						isPseudo(childSelector.parent) &&
+						isContextFunctionalPseudoClass(childSelector.parent)
+					) {
+						checkSelector(childSelector, selector, ruleNode);
+					}
```

Comment:
I don't really like that we are now checking the parent of a node multiple times.

Before we checked that a node itself was a specific kind of pseudo and then recursively called `checkSelector` on it.

Now we are walking down the AST and then we check the parent.

This is more work, marginally more work, but still more work.

Reply:
Fair point. We can accept it for now. 👍🏼